pkg/machine/e2e/config_test.go

```diff
@@ -198,6 +198,10 @@ func (m *machineTestBuilder) run() (*machineSession, error) {
 }
 
 func runWrapper(podmanBinary string, cmdArgs []string, timeout time.Duration, wait bool) (*machineSession, error) {
+	if len(cmdArgs) >= 2 && cmdArgs[0] == "machine" && cmdArgs[1] == "init" && !slices.Contains(cmdArgs, "-v") {
+		tailArgs := append([]string{"-v", ""}, cmdArgs[2:]...)
+		cmdArgs = append([]string{"machine", "init"}, tailArgs...)
+	}
```

Comment:
This does not look right. First of all this argument manipulation happens in the wrong place. It would need to happen for buildCmd() in the machineInit type.

Second I really really think tests must use the default volumes, it is critical that these mounts work. Most tests may not use them but it is still important that our code works and does not contain flakes etc.. (https://github.com/containers/podman/pull/23118) And yes this one test there actually depend on the default mounts.

Reply:
This part is indeed dirty and hacky. I need to get more familiar with the testing framework setup to improve this.

I noticed numerous tests, which were designed specifically for testing volume mounts, so, I didn't expect there will be some test, which implicitly test default ones. Probably the behavior from that test could be adjusted by loading the config as an app would do and manually converting it into chain of `-v "/some:/some" -v "/other:/other"`, but I'm not sure if it worth doing or if it is better to keep the things as is for testing as they have proved to be working fine already.

Alternatively I could keep everything as is and try to add dedicated tests for the new API option checking that it will create no mounts. And if at later point the adjustment of default testing setting would be needed it can be done in a separate PR. WDYT?

Reply:
I find it important that we test defaults. Adding explicit machine mounts means we are no longer testing how most users use it.

Reply:
Removed this workaround. Also, added a dedicated test for this command line.